src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/CommunityService.kt

```diff
@@ -0,0 +1,63 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service
+
+import com.fasterxml.jackson.annotation.JsonFormat
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.ApplicationListener
+import org.springframework.stereotype.Service
+import org.springframework.web.reactive.function.client.WebClient
+import reactor.core.publisher.Mono
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.events.ReferralEvent
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.events.ReferralEventType
+import java.time.LocalDate
+
+@Service
+class CommunityService(
+  @Value("\${communityapi.temporaryReferralRequest.probationAreaCode}") private val probationAreaCode: String,
+  @Value("\${communityapi.temporaryReferralRequest.referralType}") private val referralType: String,
+  @Value("\${communityapi.temporaryReferralRequest.staffCode}") private val staffCode: String,
+  @Value("\${communityapi.temporaryReferralRequest.teamCode}") private val teamCode: String,
+  @Value("\${interventions-ui.baseurl}") private val interventionUiBaseUrl: String,
+  private val communityApiWebClient: WebClient,
+) : ApplicationListener<ReferralEvent> {
+
+  override fun onApplicationEvent(event: ReferralEvent) {
+    when (event.type) {
+      ReferralEventType.SENT -> {
+        val url = "$interventionUiBaseUrl/referrals/${event.referral.id}/confirmation"
+        val body = ReferRequest(
+          probationAreaCode,
+          referralType,
+          staffCode,
+          teamCode,
+          url,
+          event.referral.sentAt?.toLocalDate()
```

Comment:
if we are saying this can't be null, let's make a non-null assertion, this way if it _is_ null, the code blows up

`event.referral.sentAt!!.toLocalDate()`

of course we need to ensure we are handling the potential exception though